AutoSteamApp/Program.cs

```diff
@@ -1,472 +1,73 @@
-        private static KeystrokeAPI api = new KeystrokeAPI();
-        private static readonly Dictionary<VirtualKeyCode, int> keyOrder = new Dictionary<VirtualKeyCode, int>()
-        {
-            { VirtualKeyCode.VK_A, 999 },
-            { VirtualKeyCode.VK_W, 999 },
-            { VirtualKeyCode.VK_D, 999 },
-            { VirtualKeyCode.VK_Q, 999 },
-            { VirtualKeyCode.VK_Z, 999 },
-        };
-
-        private static readonly Dictionary<int, List<int>> rndPatterns = new Dictionary<int, List<int>>()
-        {
-            { 0, new List<int> { 0, 1, 2 } },
-            { 1, new List<int> { 1, 0, 2 } },
-            { 2, new List<int> { 2, 0, 1 } },
-            { 3, new List<int> { 0, 2, 1 } },
-            { 4, new List<int> { 2, 1, 0 } },
-            { 5, new List<int> { 1, 2, 0 } }
-        };
-
-        private static Process mhw;
-        private static CancellationTokenSource ct = new CancellationTokenSource();
-        private static bool IsCorrectVersion = false;
-        private static bool IsSmartRun = false;
+        #region Main
 
         static void Main(string[] args)
         {
-            WriteMenu();
 
-            HookKeyboardEvents();
-
-            Startup();
-
-            if (IsCorrectVersion)
-            {
-                DoWork(IsSmartRun);
-            }
-            else
+            // Attempts to load an external config file if one is supplied
+            if (args.Length > 0)
             {
-                DoRandomWork();
+                ConfigurationReader.LoadConfig(args[0]);
             }
-        }
-
-        private static void WriteMenu()
-        {
-            Console.Title = $"Currently built for version: ({Settings.SupportedGameVersion})";
-            Console.WriteLine($"Currently built for version: {Settings.SupportedGameVersion}");
 
-            Console.WriteLine(string.Empty);
+            // Set the console title
+            StaticHelpers.SetConsoleTitle();
 
-            Console.WriteLine(
-                string.Format(
-                    "Based on the current settings, this run will consume: {0} fuel. If this was not intended, please change AutoSteamApp.exe.config.",
-                    Settings.ShouldConsumeAllFuel ? "ALL the available" : "ONLY the Natural"));
-            Console.WriteLine(string.Empty);
+            // Initialize the logger
+            StaticHelpers.SetLogger();
 
-            WriteSeparator();
-            Console.WriteLine($"Please select the type of run you want. When the run is finished, the app will close.");
+            // Print config settings
+            StaticHelpers.DisplayConfig();
 
-            WriteSeparator();
-            Console.WriteLine($"Press '{((KeyCode)Settings.KeyCodeStart).ToString()}' to ->");
-            Console.WriteLine($"        Run with 100% Accuracy (requires correct game version)");
+            // Create the automaton
+            SteamworkAutomaton automaton = new SteamworkAutomaton();
```

Comment:
The whole instance of `SteamworkAutomaton` depends on whether the CancellationToken is canceled or not - so i would pass the token in the constructor (or some other way of DI). Also, anything that is needed by SteamworkAutomaton should `injected` in.